src/Compilers/CSharp/Portable/Declarations/MergedTypeDeclaration.cs

```diff
@@ -123,6 +123,22 @@ public bool AnyMemberHasAttributes
             }
         }
 
+        public bool HasConstraints
+        {
+            get
+            {
+                foreach (var decl in this.Declarations)
```

Comment:
Consider `return this.Declarations.Any(d => d.HasConstraints);`. I think there is an extension method on `ImmutableArray` for this (although for some reason, I always have trouble to find it).